tests/tests_lite/plugins/collectives/test_torch_collective.py

```diff
@@ -185,61 +188,88 @@ def test_repeated_create_and_destroy():
         collective.create_group().teardown()
 
 
-def collective_launch(fn, parallel_devices, num_groups=1):
+def collective_launch(fn, device_type, num_devices=1, autosetup_strategy=False, num_groups=1):
     device_to_accelerator = {"cuda": CUDAAccelerator, "cpu": CPUAccelerator}
-    accelerator_cls = device_to_accelerator[parallel_devices[0].type]
+    accelerator_cls = device_to_accelerator[device_type]
+    devices = [torch.device(f"{device_type}:{i}") for i in range(num_devices)]
     strategy = DDPSpawnStrategy(
-        accelerator=accelerator_cls(), parallel_devices=parallel_devices, cluster_environment=LightningEnvironment()
+        accelerator=accelerator_cls(), parallel_devices=devices, cluster_environment=LightningEnvironment()
     )
     launcher = _MultiProcessingLauncher(strategy=strategy)
     collectives = [TorchCollective() for _ in range(num_groups)]
-    wrapped = partial(wrap_launch_function, fn, strategy, collectives)
+    wrapped = partial(wrap_launch_function, fn, strategy, collectives, devices, autosetup_strategy)
     return launcher.launch(wrapped, strategy, *collectives)
 
 
-def wrap_launch_function(fn, strategy, collectives, *args, **kwargs):
-    strategy._set_world_ranks()
-    collectives[0].setup(  # only one needs to setup
-        world_size=strategy.num_processes,
-        main_address="localhost",
-        backend=strategy._get_process_group_backend(),
-        rank=strategy.global_rank,
-    )
-    with check_destroy_group():  # manually use the fixture for the assertions
+def wrap_launch_function(fn, strategy, collectives, devices, autosetup_strategy, *args, **kwargs):
+    with check_destroy_group(autosetup_strategy):  # manually use the fixture for the assertions
+        if autosetup_strategy:
+            strategy.setup_environment()
+        else:
+            strategy._set_world_ranks()
+            collectives[0].setup(  # only one needs to setup
+                world_size=strategy.num_processes,
+                main_address=strategy.cluster_environment.main_address,
+                backend=strategy._get_process_group_backend(),
+                rank=strategy.global_rank,
+                timeout=torch_collective.default_pg_timeout,
+            )
+
+        fn_params = inspect.signature(fn).parameters
+        if "device" in fn_params:
+            kwargs["device"] = devices[strategy.global_rank]
```

Comment:
This was added so that the actual function under test, that is being launched knows, which device it is supposed to be running on and therefore where it should be creating tensors.